internal/beater/beater.go

```diff
@@ -206,9 +206,11 @@ func (s *Runner) Run(ctx context.Context) error {
 			// to have a margin of safety for other processes. The fraction value
 			// of 0.625 is used to keep the 80% of the total system memory limit
 			// to be 50% of the total for calculating the number of decoders.
-			memLimitGB = float64(limit) / 1024 / 1024 / 1024 * 0.625
+			memLimit = uint64(float64(limit) * 0.625)
 		}
 	}
+	// Convert the memory limit to gigabytes to calculate the config values.
+	var memLimitGB = float64(memLimit) / 1073741824
```

Comment:
nit: the magic number is slightly hard to read
```suggestion
	var memLimitGB = float64(memLimit) / (1<<30)
```

Reply:
updated